src/hotspot/cpu/x86/stubGenerator_x86_64.hpp

```diff
@@ -575,6 +578,10 @@ class StubGenerator: public StubCodeGenerator {
 
   void generate_libm_stubs();
 
+#ifdef COMPILER2
+  void generate_string_indexof_stubs(address *fnptrs, StrIntrinsicNode::ArgEncoding ae);
```

Comment:
Is it possible to make `generate_string_indexof_stubs()` as local static method in `stubGenerator_x86_64_string.cpp` and pass `StubGenerator*` as argument?
Then you don't to include "opto/intrinsicnode.hpp" here.

Reply:
Done.